test/test-rotatesTetrominos.js

```diff
@@ -1,32 +1,194 @@
 var assert = require('chai').assert;
 var tetris = require('../tetris');
-describe("rotate L-tetromino", function (){
+describe("Class tetramino", function (){
     class Tetromino {
-        constructor(tetramino, phase, coordinates){
-            this.typeTetramoni = tetramino
-            this.initialRotationPhase = phase;
-            this.initialCoordinates = coordinates;
+        constructor(typeTetramoni, initialRotationPhase, initialCoordinates){
+            this.typeTetramoni = typeTetramoni
+            this.initialRotationPhase = initialRotationPhase;
+            this.initialCoordinates = initialCoordinates;
         }
-        _rotateTetromino(){
-					
+        rotateTetromino(typeOfTetromino, rotationPhaseOfTetromino, tetrominoCoordinates) {
+            let shift = this.determineShift(typeOfTetromino, rotationPhaseOfTetromino)
+            let pivot = this.calculatorCoordinateOfTopLeft(tetrominoCoordinates);
+            let newCoordinates = this.rotatesArrayOfCoordinates(tetrominoCoordinates, pivot)
+            let result = this.shiftCoordinates(newCoordinates, shift)
+            return result.sort(this.orderCoordinates);
         }
-        _incrementPhase(){
-            
-        }
-        _finelRotationTetromino(field){
 
+        incrementPhase (typeOfTetromino, currentRotationPhaseOfTetromino) {
+            let arrayPhases = [];
+            if (typeOfTetromino === "I" || typeOfTetromino === "S" || typeOfTetromino === "Z") {
+                arrayPhases = ["vertical", "horizontal"];
+            }
+            if (typeOfTetromino === "L" || typeOfTetromino === "J" || typeOfTetromino === "T") {
+                arrayPhases = ["up", "left", "down", "right"];
+            }
+            if (typeOfTetromino === "O") {
+                arrayPhases = ["stable"];
+            }
+            let numberPhase = arrayPhases.indexOf(currentRotationPhaseOfTetromino)
+            if (numberPhase !== arrayPhases.length - 1) {
+                return arrayPhases[numberPhase + 1]
+            } else {
+                return arrayPhases[0]
+            }
         }
-            
+
+        determineShift (typeOfTetromino, rotationPhaseOfTetromino) {
+            if (typeOfTetromino === "O") {
```

Comment:
what's all this? you already have this functionality in tetris.determineShift

Reply:
I do not understand, can I use methods from a tetris file?